test/Test/Toml/Type/PrefixTree.hs

```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE PatternSynonyms #-}
+
+module Test.Toml.Type.PrefixTree
+    ( prefixTreeSpec
+    ) where
+
+import Hedgehog (forAll, (===))
+import Test.Hspec (Arg, Expectation, Spec, SpecWith, describe, it, shouldBe)
+import Test.Hspec.Hedgehog (hedgehog)
+
+import Test.Toml.Gen (genKey, genPrefixMap, genVal)
+import Test.Toml.Property (assocSemigroup, leftIdentityMonoid, rightIdentityMonoid)
+import Toml.Type.Key (pattern (:||))
+
+import qualified Toml.Type.PrefixTree as Prefix
+
+
+prefixTreeSpec :: Spec
+prefixTreeSpec = describe "PrefixTree unit and property tests" $ do
+    prefixTreeUnitSpec
+    prefixTreePropertySpec
+
+prefixTreeUnitSpec :: Spec
+prefixTreeUnitSpec = describe "Unit tests for basic cases" $ do
+    -- some test keys
+    let a  = "a" :|| []
+    let b  = "b" :|| []
+    let c  = "c" :|| []
+    let ab = "a" :|| ["b"]
+
+    describe "Insert and lookup unit tests" $ do
```

Comment:
I guess you can get rid of two levels of `describe` in here and make one instead